diesel/src/query_builder/upsert/on_conflict_target_decorations.rs

```diff
@@ -0,0 +1,69 @@
+use crate::backend::{Backend, SupportsOnConflictClause, SupportsOnConflictTargetDecorations};
+use crate::expression::Expression;
+use crate::query_builder::upsert::on_conflict_target::{ConflictTarget, NoConflictTarget};
+use crate::query_builder::where_clause::{NoWhereClause, WhereAnd, WhereClause};
+use crate::query_builder::{AstPass, QueryFragment, QueryResult};
+use crate::sql_types::BoolOrNullableBool;
+
+pub trait UndecoratedConflictTarget {}
+
+impl UndecoratedConflictTarget for NoConflictTarget {}
+impl<T> UndecoratedConflictTarget for ConflictTarget<T> {}
+
+/// Interface to add information to conflict targets.
+/// Designed to be open for further additions to conflict targets like constraints
+pub trait DecoratableTarget<P> {
+    /// Output type of filter_target operation
+    type FilterOutput;
+    /// equivalent to filter of FilterDsl but aimed at conflict targets
+    fn filter_target(self, predicate: P) -> Self::FilterOutput;
+}
+
+#[derive(Debug)]
+pub struct DecoratedConflictTarget<T, U> {
+    target: T,
+    where_clause: U,
+}
+
+impl<T, P> DecoratableTarget<P> for T
+where
+    P: Expression,
+    P::SqlType: BoolOrNullableBool,
+    T: UndecoratedConflictTarget,
+{
+    type FilterOutput = DecoratedConflictTarget<T, WhereClause<P>>;
+    fn filter_target(self, predicate: P) -> Self::FilterOutput {
+        DecoratedConflictTarget {
+            target: self,
+            where_clause: NoWhereClause.and(predicate),
+        }
+    }
+}
+
+impl<T, U, P> DecoratableTarget<P> for DecoratedConflictTarget<T, U>
+where
+    P: Expression,
+    P::SqlType: BoolOrNullableBool,
+    U: WhereAnd<P>,
+{
+    type FilterOutput = DecoratedConflictTarget<T, <U as WhereAnd<P>>::Output>;
```

Comment:
```suggestion
    type FilterOutput = DecoratedConflictTarget<T, <U as WhereAnd<P>>::Output>;
    
```

Reply:
what did you want to express? the original line is identical with the proposed one.

Reply:
done